tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -1789,7 +1789,7 @@ void SpirvEmitter::doVarDecl(const VarDecl *decl) {
     return;
   }
 
-  if (isConstantTextureBuffer(decl->getType())) {
+  if (isConstantTextureBuffer(decl->getType()) && !decl->hasInit()) {
```

Comment:
Why is `hasInit` the correct condition? Your code seems to fail with this example https://godbolt.org/z/WGjTzjc61.

Reply:
This might be the better condition to check:

https://github.com/microsoft/DirectXShaderCompiler/blob/15c87234584676574ca4a4ebebde03aab9822994/tools/clang/lib/SPIRV/SpirvEmitter.h#L97-L105

Reply:
Yeah, that's much better, thanks